terra_opp/renderers.py

```diff
@@ -84,12 +84,13 @@ class ZipRenderer(renderers.BaseRenderer):
     format = 'zip'
 
     def render(self, data, media_type=None, renderer_context=None):
-        with tempfile.SpooledTemporaryFile() as tmp:
-            with zipfile.ZipFile(tmp, 'w', zipfile.ZIP_DEFLATED) as archive:
-                for file in data:
-                    archive.writestr(
-                        os.path.basename(file.name),
-                        file.open().read(),
-                    )
-            tmp.seek(0)
-            return tmp.read()
+        if isinstance(data, list):
+            with tempfile.SpooledTemporaryFile() as tmp:
+                with zipfile.ZipFile(tmp, 'w', zipfile.ZIP_DEFLATED) as archive:
+                    for file in data:
+                        archive.writestr(
+                            os.path.basename(file.name),
+                            file.open().read(),
+                        )
+                tmp.seek(0)
+                return tmp.read()
```

Comment:
you need to keep return result in your response. maybe by returning parent with return super() ?

Reply:
That's what I've done in the new push.